runtime/moonbase/src/lib.rs

```diff
@@ -1091,6 +1092,95 @@ impl pallet_moonbeam_orbiters::Config for Runtime {
 	type WeightInfo = pallet_moonbeam_orbiters::weights::SubstrateWeight<Runtime>;
 }
 
+/// Only callable after `set_validation_data` is called which forms this proof the same way
+fn relay_chain_state_proof() -> RelayChainStateProof {
+	let relay_storage_root = ParachainSystem::validation_data()
+		.expect("set in `set_validation_data`")
+		.relay_parent_storage_root;
+	let relay_chain_state =
+		ParachainSystem::relay_state_proof().expect("set in `set_validation_data`");
+	RelayChainStateProof::new(ParachainInfo::get(), relay_storage_root, relay_chain_state)
+		.expect("Invalid relay chain state proof, already constructed in `set_validation_data`")
+}
+
+pub struct BabeDataGetter;
+impl pallet_randomness::GetBabeData<BlockNumber, u64, Option<Hash>> for BabeDataGetter {
+	// Tolerate panic here because only ever called in inherent (so can be omitted)
+	fn get_relay_block_number() -> BlockNumber {
+		ParachainSystem::validation_data()
+			.expect("set in `set_validation_data`inherent => available before on_initialize")
+			.relay_parent_number
+	}
+	// Tolerate panic here because only ever called in inherent (so can be omitted)
+	fn get_relay_epoch_index() -> u64 {
+		relay_chain_state_proof()
+			.read_optional_entry(relay_chain::well_known_keys::EPOCH_INDEX)
+			.ok()
+			.flatten()
+			.expect("expected to be able to read epoch index from relay chain state proof")
+	}
+	fn get_current_block_randomness() -> Option<Hash> {
+		relay_chain_state_proof()
+			.read_optional_entry(relay_chain::well_known_keys::CURRENT_BLOCK_RANDOMNESS)
+			.ok()
+			.flatten()
+	}
+	fn get_one_epoch_ago_randomness() -> Option<Hash> {
+		relay_chain_state_proof()
+			.read_optional_entry(relay_chain::well_known_keys::ONE_EPOCH_AGO_RANDOMNESS)
+			.ok()
+			.flatten()
+	}
+	fn get_two_epochs_ago_randomness() -> Option<Hash> {
+		relay_chain_state_proof()
+			.read_optional_entry(relay_chain::well_known_keys::TWO_EPOCHS_AGO_RANDOMNESS)
+			.ok()
+			.flatten()
+	}
+}
+
+pub struct VrfInputGetter;
+impl session_keys_primitives::GetVrfInput<session_keys_primitives::VrfInput<Slot, Hash>>
+	for VrfInputGetter
+{
+	fn get_vrf_input() -> session_keys_primitives::VrfInput<Slot, Hash> {
+		session_keys_primitives::VrfInput {
+			slot_number: relay_chain_state_proof()
+				.read_slot()
+				.expect("CheckInherents reads slot from state proof QED"),
+			storage_root: ParachainSystem::validation_data()
+				.expect("set in `set_validation_data`inherent => available before on_initialize")
+				.relay_parent_storage_root,
+		}
+	}
+}
+
+pub struct ReserveAccount;
```

Comment:
Cant we do this through parameter_types?

Reply:
An alternative is the way pallets like `Treasury` do this: https://github.com/paritytech/substrate/blob/54d6f2cbae12fa314f2bcaf6d57fba585162438f/frame/treasury/src/lib.rs#L487

Reply:
deja vu https://github.com/PureStake/moonbeam/pull/680#discussion_r709112572 

I followed how @nanocryk did it in https://github.com/PureStake/moonbeam/pull/1653 , which I think works and is fine 
 -- it is Config associated type `Get<AccountId` which allows to configure the address in the runtime